### ISSUE TITLE
Questions Regarding Preprocessing Implementation

### ISSUE THREAD

**Issue by OrelZang** [dnguyengithub/MultitaskAIS#20]
Hi @dnguyengithub ,

I'm trying to implement your proposed solution on another AIS dataset but I'm facing a few problems in the preprocessing steps:

- I'm having a hard time understanding this whole sampling process. In the paper you present the problem that we need regularly sampled data to train the model first. However, I am not sure I fully understand your proposed solution (i.e. using  constant velocity model to create regularly time-sampled AIS tracks). Moreover, if I understand your solution correctly, it sounds rather odd to generate regularly sampled data in such an artificial manner. I would really appreciate some explanation on this topic.

- I'm a bit concerned about your train-test splitting strategy. In the code I can see that you split your data based on random permutation rather based on continuous time intervals. Doesn't it beat the purpose of learning sequences patterns from the past and check if they are valid in the future?

- In the paper you mention that you only chose tracks which the maximum time interval between two successive received AIS messages is 1 hour. However, in your code I see that you split the tracks using 2 hours interval. What is the correct interval value?

- In the paper you mention that you created regulary time-sampled AIS tracks at 10 minutes time scale. However, in your code I see that you used a 5 minutes sampling resolution. What is the correct interval value?

Any help with one or all of these questions will be much appreciated.

Thanks in advance.

**Comment by duonginspace** [comment]
Hi,

1. Yes, we used geodetic interpolation to geodesic interpolation to resample the track regularly. (see https://pyproj4.github.io/pyproj/dev/_modules/pyproj/geod.html). As long as the interval between two consecutive AIS messages is small, the error inherited from this interpolation wont be a problem.

2. Yes, this train-test splitting strategy is wrong. We should use data from one period for the training set then data from the next period for the test set. We tried the correct train/test split and the results are still good. We wrote another paper on this problem, I will update this repository as soon as our new paper is accepted.

3. The results in the papers were obtained with max_interval = 1 hour. However, for the Brittany dataset, max_interval = 1 hour and max_interval = 2 hours give similar results. 

4.  The sampling resolution is 10 mins. (see line #90 in datasets.py)

Hope that this answers your questions.
